ambry-network/src/main/java/com/github/ambry/network/http2/Http2BlockingChannel.java

```diff
@@ -68,118 +45,78 @@
 
 
 /**
- * A HTTP2 implementation of {@link ConnectedChannel}. This implementation is for test now. It will be imporved to
- * to support replication in the future.
+ * A HTTP2 implementation of {@link ConnectedChannel} used in replication and test.
+ * This implementation reuses connections maintained by {@link Http2MultiplexedChannelPool}(a pool for a single host:port) and creates a stream channel for each send.
  */
 public class Http2BlockingChannel implements ConnectedChannel {
   private static final Logger logger = LoggerFactory.getLogger(Http2BlockingChannel.class);
   private final static AttributeKey<Promise<ByteBuf>> RESPONSE_PROMISE = AttributeKey.newInstance("ResponsePromise");
-  private final String hostName;
-  private final int readBufferSize;
-  private final int writeBufferSize;
-  private final int http2InitialWindowSize;
-  private final SSLFactory sslFactory;
-  private final int port;
-  private EventLoopGroup workerGroup;
-  private Channel channel;
   private Promise<ByteBuf> responsePromise;
-  private Http2StreamChannelBootstrap http2StreamChannelBootstrap;
+  private Http2MultiplexedChannelPool http2MultiplexedChannelPool;
 
-  public Http2BlockingChannel(String hostName, int port, SSLConfig sslConfig, int readBufferSize, int writeBufferSize,
-      int http2InitialWindowSize) {
-    this.hostName = hostName;
-    this.port = port;
-    this.readBufferSize = readBufferSize;
-    this.writeBufferSize = writeBufferSize;
-    this.http2InitialWindowSize = http2InitialWindowSize;
+  public Http2BlockingChannel(Http2MultiplexedChannelPool http2MultiplexedChannelPool) {
+    this.http2MultiplexedChannelPool = http2MultiplexedChannelPool;
+  }
+
+  /**
+   * Constructor for test purpose.
+   */
+  public Http2BlockingChannel(String hostName, int port, SSLConfig sslConfig, Http2ClientConfig http2ClientConfig,
+      Http2ClientMetrics http2ClientMetrics) {
+    NettySslHttp2Factory nettySslHttp2Factory;
     try {
-      sslFactory = new NettySslHttp2Factory(sslConfig);
+      nettySslHttp2Factory = new NettySslHttp2Factory(sslConfig);
     } catch (GeneralSecurityException | IOException e) {
-      throw new IllegalStateException(e);
+      throw new IllegalStateException("Can't create NettySslHttp2Factory: ", e);
     }
+    this.http2MultiplexedChannelPool =
+        new Http2MultiplexedChannelPool(new InetSocketAddress(hostName, port), nettySslHttp2Factory,
```

Comment:
nit: let's keep hostname and port so in the logger and exception we can print host and port out.